pkg/pod/pod.go

```diff
@@ -290,7 +290,11 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 	}
 
 	podAnnotations := taskRun.Annotations
-	podAnnotations[ReleaseAnnotation] = version.PipelineVersion
+	version, err := changeset.Get()
+	if err != nil {
+		return nil, err
+	}
+	podAnnotations[ReleaseAnnotation] = version
```

Comment:
Maybe we should log something and print `unknown` here instead of erroring out, wdyt @mattmoor ?